File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y
  | notWithin 10 = 0
  | notWithin 5  = 1
  | notWithin 1  = 5
  | otherwise    = 10
  where
    dist = sqrt ((x * x) + (y * y))
    notWithin d = dist > d

File: solutions/cpp/darts/1/darts.cpp
#include "darts.h"
#include <cmath>

namespace darts {

int score(float x, float y){
    float hyp = sqrt(pow(x,2) + pow(y,2));
    int score;
    if (hyp <= 1) {score = 10;}
    else if (hyp <= 5) {score = 5;}
    else if (hyp <= 10) {score = 1;}
    else {score = 0;}
    return score;
}

}  // namespace darts
